xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp

```diff
@@ -209,10 +209,10 @@ void CRendererDRMPRIME::SetVideoPlane(CVideoBufferDRMPRIME* buffer)
       return;
     }
 
-    int32_t crtc_x = (int32_t)m_destRect.x1;
-    int32_t crtc_y = (int32_t)m_destRect.y1;
-    uint32_t crtc_w = (uint32_t)m_destRect.Width();
-    uint32_t crtc_h = (uint32_t)m_destRect.Height();
+    int32_t crtc_x = static_cast<int32_t>(m_destRect.x1) & ~1;
+    int32_t crtc_y = static_cast<int32_t>(m_destRect.y1) & ~1;
+    uint32_t crtc_w = (static_cast<uint32_t>(m_destRect.Width()) + 1) & ~1;
```

Comment:
Whats the reason for the +1 ?

Reply:
It is used to round up, we can probably just round both x/y and w/h down but wanted to keep it in line with `mpv`, should I add change it to use a macro similar to https://github.com/mpv-player/mpv/blob/master/video/out/opengl/hwdec_drmprime_drm.c#L178-L179 ?

Reply:
It's all good

Reply:
Basically it does not round up. It makes sure that the number itself here is dividable by 2.
10 + 1 -> 10 
11 + 1 -> 100 -> 100

So numbers not dividable by two get +1 added so that they are afterwards.

Reply:
Yeah you are correct, I meant to say align up not round up, uneven width/height gets aligned up to closest number dividable by 2 and uneven x/y gets aligned down to closest number dividable by 2.

Reply:
Perfect :-) It's just that bitshiftery always gets my attention.